### ISSUE TITLE
[Panic] index out of bounds on micropython/micropython

### ISSUE THREAD

**Issue by cclauss** [charliermarsh/ruff#3404]
```
ruff --version  # --> ruff 0.0.254
git clone https://github.com/micropython/micropython
cd micropython
RUST_BACKTRACE=1 ruff .
```
% `RUST_BACKTRACE=1 ruff --exclude=tests . ` # works as expected.
```
error: `ruff` crashed. This indicates a bug in `ruff`. If you could open an issue at:

https://github.com/charliermarsh/ruff/issues/new?title=%5BPanic%5D

quoting the executed command, along with the relevant file contents and `pyproject.toml` settings, we'd be very appreciative!

thread '<unnamed>' panicked at 'index out of bounds: the len is 1 but the index is 1', crates/ruff/src/checkers/noqa.rs:42:38
stack backtrace:
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
[1]    38956 abort      RUST_BACKTRACE=1 ruff .
```

**Comment by charliermarsh** [comment]
Thanks! Will take a look today.

**Comment by charliermarsh** [comment]
It's panicking on this test file which contains CR line endings: `micropython/tests/basics/string_cr_conversion.py`.

(\cc @MichaReiser for visibility, think you had a comment somewhere around "continuation on panic".)

**Comment by charliermarsh** [comment]
As good a time as any to implement graceful fallback for CR line endings.

**Comment by charliermarsh** [comment]
Resolved via #3454 and #3439.
